devices/src/legacy/rtc_pl031.rs

```diff
@@ -0,0 +1,264 @@
+            match_value: 0,
+            load: 0,
+            imsc: 0,
+            ris: 0,
+            interrupt_evt,
+        }
+    }
+
+    fn trigger_interrupt(&mut self) -> Result<()> {
+        self.interrupt_evt.write(1).map_err(Error::InterruptFailure)
+    }
+
+    fn get_time(&self) -> u32 {
+        let ts = self.tick_offset.add(SteadyTime::now() - self.previous_now);
+        ts.sec as u32
+    }
+
+    fn handle_write(&mut self, offset: u64, v: u32) -> Result<()> {
+        match offset {
+            MR => {
+                // The MR register is used for implementing the RTC alarm. A real time clock alarm is
+                // a feature that can be used to allow a computer to 'wake up' after shut down to execute
+                // tasks every day or on a certain day. It can sometimes be found in the 'Power Management'
+                // section of a motherboard's BIOS setup. This is functionality that extends beyond
+                // Firecracker intended use. However, we increment a metric just in case.
+                self.match_value = v;
+                METRICS.rtc.missed_write_count.inc();
+            }
+            LR => {
+                self.load = v;
+                self.previous_now = SteadyTime::now();
+                self.tick_offset = Timespec::new(i64::from(v), 0);
+            }
+            IMSC => {
+                self.imsc = v & 1;
+                self.trigger_interrupt()?;
+            }
+            ICR => {
+                // As per above mentioned doc, the interrupt is cleared by writing any data value to
+                // the Interrupt Clear Register.
+                self.ris = 0;
+                self.trigger_interrupt()?;
+            }
+            CR => (), // ignore attempts to turn off the timer.
+            o => {
+                return Err(Error::BadWriteOffset(o));
+            }
+        }
+        Ok(())
+    }
+}
+
+impl BusDevice for RTC {
+    fn read(&mut self, offset: u64, data: &mut [u8]) {
+        let v;
+        let mut read_ok = true;
+
+        if offset < AMBA_ID_HIGH && offset >= AMBA_ID_LOW {
+            let index = ((offset - AMBA_ID_LOW) >> 2) as usize;
+            v = u32::from(PL031_ID[index]);
+        } else {
+            v = match offset {
+                DR => self.get_time(),
+                MR => {
+                    METRICS.rtc.missed_read_count.inc();
+                    // Even though we are not implementing RTC alarm we return the last value
+                    self.match_value
+                }
+                LR => self.load,
+                CR => 1, // RTC is always enabled.
+                IMSC => self.imsc,
+                RIS => self.ris,
+                MIS => self.ris & self.imsc,
+                _ => {
+                    read_ok = false;
+                    0
+                }
+            };
+        }
+        if read_ok {
+            LittleEndian::write_u32(data, v);
+        } else {
+            error!(
```

Comment:
Indeed, error might be too much. I do think though that it is more than an info since not having the  time updated could lead to other serious problems. So, I left it as a warning.